Modelica/Blocks/Logical.mo

```diff
@@ -929,6 +929,34 @@ The output <code>Q</code> is set by the input <code>S</code>, is reset by the in
 </html>"));
   end RSFlipFlop;
 
+  block AssertCondition "Asserts that input condition u is true"
+    extends Modelica.Blocks.Interfaces.partialBooleanSI;
+    parameter String message = getInstanceName() + "'s input is false"
+      "Assertion message to be displayed if the input u is false";
```

Comment:
To me this use of getInstanceName is problematic for two reasons:
* It is redundant and encourages work-arounds for something that should be handled by tools.
* It seems complicated and thus users might want to leave it as is, instead of actually setting the message to something meaningful.

For the first point Dymola currently writes the following if it fails:
```
Error: The following error was detected at time: 0.5
  M2.m.assertCondition's input is false
  Failed condition: m.assertCondition.u
```
So the `getInstanceName()` doesn't add anything.

For the second item I instead modified the test-case to add a Xor connected to the input _and_ output of a not-gate and set the message:
`message="The input and output of Not shall be different"`

That specific test might not be the best, but I believe that something along those lines would better show how it is intended to be used - especially as using asserts in test-cases is a well-known practice in test-driven development.

So I would prefer something simpler:
```suggestion
    parameter String message = "Assert block failed"
      "Assertion message to be displayed if the input u is false; please adapt";
```

Reply:
I am also in favor of not having `getInstanceName()` as part of the message.

(By the way, I guess there is no way that the user can restore the effect of a default `getInstanceName()` with the use of a modification, as any modification is going to be resolved as coming from some other component.)